scripts/get-nightly-build-pass-rate.ts

```diff
@@ -0,0 +1,108 @@
+import puppeteer from 'puppeteer';
+import dotenv from 'dotenv';
+dotenv.config();
+
+const loginAndAnalyzeBuilds = async (
+  latestBuildNumber: number,
+  numberOfDays: number,
+  username: string,
+  token: string
+) => {
+  const loginUrl = process.env.JENKINS_LOGIN_URL;
+  const baseApiUrl = process.env.JENKINS_BASE_API_URL;
+  const browser = await puppeteer.launch({ headless: true });
+  const page = await browser.newPage();
+
+  if (!loginUrl || !baseApiUrl) {
+    console.error('Missing environment variables. Please check your .env file.');
+    return;
+  }
+
+  try {
+    console.log('Logging in to Jenkins...');
+    await page.goto(loginUrl, { waitUntil: 'networkidle2' });
+    await page.type('#j_username', username);
+    await page.type('input[name="j_password"]', token);
+    await Promise.all([
+      page.click('button[type="submit"]'),
+      page.waitForNavigation({ waitUntil: 'networkidle2' }),
+    ]);
+    console.log('Login successful.');
+
+    const startBuildNumber = latestBuildNumber - numberOfDays + 1;
+    console.log(`Analyzing builds from ${startBuildNumber} to ${latestBuildNumber}...`);
+
+    const failedBuilds: number[] = [];
+    let passCount = 0;
+
+    for (let buildNumber = startBuildNumber; buildNumber <= latestBuildNumber; buildNumber++) {
+      console.log(`Fetching data for build #${buildNumber}...`);
+      const url = `${baseApiUrl}/${buildNumber}/api/json`;
+      try {
+        const response = await page.goto(url, { waitUntil: 'networkidle0', timeout: 10000 });
+
+        if (!response) {
+          console.log(`Build #${buildNumber}: No response from server.`);
+          continue;
+        }
+
+        const responseBody = await response.text();
+
+        if (!responseBody) {
+          console.log(`Build #${buildNumber}: Empty response body.`);
+          continue;
+        }
+
+        let data: { result?: string } | null = null;
+        try {
```

Comment:
Planning to move the block that handles parsing and processing build data into a separate function in the future. This will make the code more modular and easier to maintain.

Reply:
Nit: This seems pretty straightforward and not super long, I think you could leave it inlined but a separate function is fine too!